Electron/src/RAM/Autonomy/components/StateDiagram.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { Component } from "react"
+import CSS from "csstype"
+import { rovecomm } from "../../../Core/RoveProtocol/Rovecomm"
+
+const container: CSS.Properties = {
+  display: "flex",
+  flexDirection: "column",
+  fontFamily: "arial",
+  borderTopWidth: "28px",
+  borderColor: "#990000",
+  borderBottomWidth: "2px",
+  borderStyle: "solid",
+  padding: "5px",
+  alignItems: "center",
+  height: "calc(100% - 50px)",
+}
+const label: CSS.Properties = {
+  marginTop: "-10px",
+  position: "relative",
+  top: "24px",
+  left: "3px",
+  fontFamily: "arial",
+  fontSize: "16px",
+  zIndex: 1,
+  color: "white",
+}
+
+interface IProps {
+  style?: CSS.Properties
+}
+
+interface IState {}
+
+const stateEnum = {
+  Idle: 0,
+  Navigating: 1,
+  SearchPattern: 2,
+  ApproachingMarker: 3,
+}
+
+class StateDiagram extends Component<IProps, IState> {
+  canvasRef: any
+
+  constructor(props: IProps) {
+    super(props)
+    this.state = {}
+
+    this.canvasRef = React.createRef()
+    rovecomm.on("CurrentState", (data: any) => this.updateStateDiagram(data))
+  }
+
+  componentDidMount() {
```

Comment:
i'm confused lol. what does this function do?

Reply:
componentDidMount is a built in function that gets called whenever the component successfully mounts (or is fully constructed). That ensures this function runs once (and only once) after all variables have been created and defaulted. We call this.updateStateDiagram[-1] to set the clear the state diagram.